queue/abstract/driver/utube.lua

```diff
@@ -57,13 +72,56 @@ function tube.create_space(space_name, opts)
 end
 
 -- start tube on space
-function tube.new(space, on_task_change)
+function tube.new(space, on_task_change, opts)
     validate_space(space)
 
+    local space_opts         = {}
+    space_opts.temporary     = opts.temporary or false
+    space_opts.engine        = opts.engine or 'memtx'
+    space_opts.format = {
+        {name = 'task_id', type = num_type()},
+        {name = 'utube', type = str_type()}
+    }
+
+    local space_ready_name = space.name .. "_utube_ready_buffer"
+    local space_ready = box.space[space_ready_name]
+    -- Feature implemented only for memtx engine for now.
+    -- https://github.com/tarantool/queue/issues/230.
+    if opts.storage_mode == tube.STORAGE_MODE_UTUBE_READY_SPACE and opts.engine == 'vinyl' then
+        error(string.format('"%s" storage mode cannot be used with vinyl engine',
+                tube.STORAGE_MODE_UTUBE_READY_SPACE))
+    end
+
+    local ready_space = (opts.storage_mode == tube.STORAGE_MODE_UTUBE_READY_SPACE)
+    if ready_space then
+        if not space_ready then
+            -- Create a space for first ready tasks from each utube.
+            space_ready = box.schema.create_space(space_ready_name, space_opts)
+            space_ready:create_index('task_id', {
+                type = 'tree',
+                parts = {1, num_type()},
+                unique = true
+            })
+            space_ready:create_index('utube', {
+                type = 'tree',
+                parts = {2, str_type()},
+                unique = true
+            })
+        else
+            validate_space_ready(space_ready)
+            if space:len() == 0 then
+                space_ready:truncate()
+            end
+        end
+    end
+
     on_task_change = on_task_change or (function() end)
     local self = setmetatable({
         space          = space,
+        space_ready    = space_ready,
         on_task_change = on_task_change,
+        ready_space    = ready_space,
```

Comment:
Seems more compact with already declared flag:
```lua
if self.ready_space then
    ...
end
```
instead of
```lua
if self.opts.storage_mode == tube.STORAGE_MODE_UTUBE_READY_SPACE then
    ...
end
```